### ISSUE TITLE
AppCleaner Accessibility service is enabled but not running. 

### ISSUE THREAD

**Issue by NikhilKrDubey** [d4rken/sdmaid-public#4345]
Since the latest update of  sd maid, the app cache cleaner is not able to clean cache of apps as it used to do in previous builds. I have the bought the pro version of sd maid and have been using it since a long time. Please fix with an upcoming update and make sd maid pro as easy as it was before.

**Comment by alx2000** [comment]
Same on Xiaomi.

**Comment by Duckrinium** [comment]
**Device: Redmi Note 8T, MIUI 12, Android 10**
SD Main asks for accessibility features, then after I enable it, it says:
`Service is enabled, but not running. Try re-enabling it.`
No matter how many times and how I tried, not working at all.

**Comment by d4rken** [comment]
My current understanding is that it is related to how SD Maid is closed.
If SD Maid killed by a "cleaner" type of app, then the system may think that SD Maid crashed and will not allow the accessibility service to run until reboot.

There was no change on SD Maids side towards this logic.
If someone can reproduce this and share a `logcat` file with me I might be able to understand it better.

**Comment by NikhilKrDubey** [comment]
My only concern is that SD maid is now not behaving as it used to behave earlier. It's not able to clean the app cache now since the latest update. Please fix this or make the app to behave as it used to behave earlier. I use the oro version...i have paid for the app....Pllleeaassee!!!!!

**Comment by NikhilKrDubey** [comment]
> My current understanding is that it is related to how SD Maid is closed.
> If SD Maid killed by a "cleaner" type of app, then the system may think that SD Maid crashed and will not allow the accessibility service to run until reboot.
> 
> There was no change on SD Maids side towards this logic.
> If someone can reproduce this and share a `logcat` file with me I might be able to understand it better.

My only concern is that SD maid is now not behaving as it used to behave earlier. It's not able to clean the app cache now since the latest update. Please fix this or make the app to behave as it used to behave earlier. I use the oro version...i have paid for the app....Pllleeaassee!!!!!

**Comment by alx2000** [comment]
In my case, I enable accessibility and SD Maid stays in endless loop at same app to clean and no cache is cleaned.
It seems that box positions on info app must be changed...

Xiaomi Mi 8 Android 10 MIUI 12

**Comment by d4rken** [comment]
> In my case, I enable accessibility and SD Maid stays in endless loop at same app to clean and no cache is cleaned.
It seems that box positions on info app must be changed...

> Xiaomi Mi 8 Android 10 MIUI 12

That's a different issue and should be an easy fix, please open a new ticket for that and provide a debug log.

**Comment by NikhilKrDubey** [comment]
> In my case, I enable accessibility and SD Maid stays in endless loop at same app to clean and no cache is cleaned.
> It seems that box positions on info app must be changed...
> 
> Xiaomi Mi 8 Android 10 MIUI 12

The same is happening with me as well, in my case after enabling accessibility services, when it starts to clean the cache, it stops while clearing 'media storage' cache it takes time to clean it, it finally skips it and move forward and after the process is complete, when you search for app cache again, the cache reappears again!!! It's actually not cleaning the cache of the apps since it's latest update!!! SD Maid pro app is kind of completely broken since it's latest update!!! Devs, kindly fix this issue!!!

**Comment by alx2000** [comment]
Both are the same issue. One running apps cleaner only and another running all cleaners together.

It must be happenning in all Android 10 wih MIUI 12. Xiaomi, Redmi, Poco...

**Comment by d4rken** [comment]
My current understanding is that "Accessibility service is enabled but not running" is not an SD Maid issue, but systems killing SD Maid such that it looks like the accessibility service crashed.

> If SD Maid killed by a "cleaner" type of app, then the system may think that SD Maid crashed and will not allow the accessibility service to run until reboot.

https://github.com/d4rken/sdmaid-public/wiki/AppCleaner#troubleshooting

There is not much I can do from my end. Which is why i'm closing this.

> In my case, I enable accessibility and SD Maid stays in endless loop at same app to clean and no cache is cleaned.

If SD Maid is looping on apps, then the system UI changed or is not recognized, this is something I can fix if someone helps me with the details for each setup. Please create an extra ticket per ROM/language combination.

**Comment by d4rken** [comment]
Example ticket where a user helped me add support for their ROM/language: https://github.com/d4rken/sdmaid-public/issues/3861

**Comment by d4rken** [comment]
Some interesting logout output from a `Redmi Note 7 (Lavender) in MIUI 12.0.1`  after closing SD Maid:


```
01-02 03:21:34.370  1942  1942 I PackageManager: Un-granting permission android.permission.WRITE_MEDIA_STORAGE from package eu.thedarken.sdm (protectionLevel=18 flags=0x3098be44)
01-02 03:21:34.370  1942  1942 I PackageManager: Un-granting permission android.permission.REQUEST_INSTALL_PACKAGES from package eu.thedarken.sdm (protectionLevel=66 flags=0x3098be44)
01-02 03:21:34.370  1942  1942 I PackageManager: Un-granting permission android.permission.CLEAR_APP_CACHE from package eu.thedarken.sdm (protectionLevel=18 flags=0x3098be44)
01-02 03:21:34.370  1942  1942 I PackageManager: Un-granting permission android.permission.PACKAGE_USAGE_STATS from package eu.thedarken.sdm (protectionLevel=114 flags=0x3098be44)
```

```
01-02 03:21:40.149  1942  1970 D Boost   : hostingType=service, hostingName={eu.thedarken.sdm/eu.thedarken.sdm.accessibility.core.ACCService}, callerPackage=android, isSystem=true, isBoostNeeded=false.
01-02 03:21:40.149  1942  1970 W ActivityManager: Slow operation: 509ms so far, now at startProcess: returned from zygote!
01-02 03:21:40.156  1942  1970 W ActivityManager: Slow operation: 515ms so far, now at startProcess: done updating battery stats
01-02 03:21:40.156  1942  1970 W ActivityManager: Slow operation: 515ms so far, now at startProcess: building log message
01-02 03:21:40.156  1942  1970 I ActivityManager: Start proc 3514:eu.thedarken.sdm/u0a164 for service {eu.thedarken.sdm/eu.thedarken.sdm.accessibility.core.ACCService} caller=android
```

```
AccessibilityManagerService: Ignoring non-encryption-aware service ComponentInfo{eu.thedarken.sdm/eu.thedarken.sdm.accessibility.core.ACCService}
```

**Comment by d4rken** [comment]
Known issue?

* https://issuetracker.google.com/issues/75414169
* Might be this bug that got closed due to lack of information or interest? https://issuetracker.google.com/issues/160506396
* https://issuetracker.google.com/issues/62498883
* https://issuetracker.google.com/issues/128470204
* https://issuetracker.google.com/issues/115931943
* https://issuetracker.google.com/issues/37123359
* https://issuetracker.google.com/issues/37100741

Basically all issues that sound related are "Won't fix (Obsolete)" :frowning:

**Comment by JeffHorinzon** [comment]
sad...

**Comment by d4rken** [comment]
See https://github.com/d4rken/sdmaid-public/issues/4386#issuecomment-753662688 for possible solutions on MIUI devices.
